activerecord/lib/active_record/core.rb

```diff
@@ -279,6 +279,14 @@ def arel_table # :nodoc:
         @arel_table ||= Arel::Table.new(table_name, type_caster: type_caster)
       end
 
+      # Returns an instance of <tt>Arel::Attributes::Attribute</tt> which represents the named column.
```

Comment:
I think we need to make `Arel::Attributes::Attribute` public and document it better.